timesync/clock.go

```diff
@@ -116,16 +117,23 @@ func (t *NodeClock) tick() {
 	}
 
 	layer := t.TimeToLayer(t.clock.Now())
+	if layer.Before(t.firstAwaitedLayer) {
+		t.log.With().Info("clock ticked back in time",
+			log.Stringer("layer", layer),
+			log.Stringer("last_ticked_layer", t.firstAwaitedLayer),
+		)
+		return
+	}
 
 	// close await channel for prev layers
-	for l := t.lastTickedLayer; !l.After(layer); l = l.Add(1) {
+	for l := t.firstAwaitedLayer; !l.After(layer); l = l.Add(1) {
```

Comment:
i think as a service, it should keep two variables, `lastTicked` and `minAwaited`.
`lastTicked` to detect skipping a tick
`minAwaited` to decide from which layer to signal channels.

Reply:
Re-added `lastTicked` and logs that are printed when the clock moves backwards in time or skips layers.